src/hooks/screen/useCreateChat.ts

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import SimpleToast from 'react-native-simple-toast';
+import {v4 as uuid} from 'uuid';
+
+import {getOrCreateAnonymousChannel} from '../../service/chat';
+import SignedMessageRepo from '../../service/repo/signedMessageRepo';
+import ChannelList from '../../database/schema/ChannelListSchema';
+import useLocalDatabaseHook from '../../database/hooks/useLocalDatabaseHook';
+import UserSchema from '../../database/schema/UserSchema';
+import ChannelListMemberSchema from '../../database/schema/ChannelListMemberSchema';
+import useChatUtilsHook from '../core/chat/useChatUtilsHook';
+import {GROUP_INFO} from '../core/constant';
+
+const useCreateChat = () => {
+  const [loadingCreateChat, setLoadingCreateChat] = React.useState(false);
+  const {localDb} = useLocalDatabaseHook();
+  const {goToChatScreen} = useChatUtilsHook();
+  const createSignChat = async (members: string[], selectedUser, from) => {
+    try {
+      setLoadingCreateChat(true);
+      const initChannel = await SignedMessageRepo.createSignedChat(members);
+      const chatData = await createChannelJson(initChannel, selectedUser);
+
+      const channelList = await ChannelList.fromMessageSignedAPI(chatData);
+      channelList.saveIfLatest(localDb);
+      handleMemberSchema(initChannel);
+      setLoadingCreateChat(false);
+      goToChatScreen(channelList, from);
+    } catch (e) {
+      setLoadingCreateChat(false);
+      console.log({e}, 'error create chat');
+    }
+  };
+
+  const createChannelJson = (response, selectedUser) => {
+    const channelWithMember = {...response.channel, members: response.members};
+    const targetRawJson = {
+      type: 'notification.message_new',
+      cid: response?.channel?.id,
+      channel_id: '',
+      channel_type: 'messaging',
+      channel: channelWithMember,
+      created_at: response?.channel,
+      targetName: selectedUser?.user?.name,
+      targetImage: selectedUser?.user?.image
+    };
+    const chatData = {
+      channel: response?.channel,
+      members: response?.members,
+      appAdditionalData: {
+        rawJson: targetRawJson,
+        message: '',
+        targetName: selectedUser?.user?.name,
+        targetImage: selectedUser?.user?.image
+      }
+    };
+    return chatData;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `createChannelJson` function, you are creating a `targetRawJson` object with properties like `created_at`, `targetName`, and `targetImage`. However, you are not checking if `response?.channel` or `selectedUser?.user` are defined before accessing their properties. This could potentially lead to undefined values in your `targetRawJson` object. Consider adding checks to ensure these objects exist before accessing their properties.



<!-- This is an auto-generated comment by OSS CodeRabbit -->